File: leetCode.com/typescript/linkedList/reverseLinkedList.ts
// Given the head of a singly linked list, reverse the list, and return the reversed list.

function reverseList(head: ListNode | null): ListNode | null {
    if (!head)
        return head
    let fast = head.next
    let previous = head
    previous.next = null
    while (fast) {
        const next = fast.next
        fast.next = previous
        previous = fast
        fast = next
    }
    return previous
};

File: leetCode.com/python/linkedList/reverseLinkedList.py
# Given the head of a singly linked list, reverse the list, and return the reversed list.

from typing import Optional
from classes import ListNode

def reverseList(head: Optional[ListNode]) -> Optional[ListNode]:
    if not head:
        return head
    fast = head.next
    previous = head
    previous.next = None
    while fast:
        next = fast.next
        fast.next = previous
        previous = fast
        fast = next
    return previous
            